autograd.h

```diff
@@ -36,10 +43,39 @@ using Tensor = tag::Tensor;
 using Container = std::shared_ptr<ContainerImpl>;
 using Optimizer = std::shared_ptr<OptimizerImpl>;
 
-void backward(Variable loss, bool keep_graph=false);
 void backward(Tensor loss, bool keep_graph=false);
-void save(std::string fn, Container const model);
-void load(std::string fn, Container model);
+
+// Some convenience functions for saving and loading
+template <typename T>
+void save(std::ostream& stream, T const & obj) {
+  cereal::BinaryOutputArchive archive(stream);
+  archive(*obj);
```

Comment:
Should be `archive(obj)`, no?

Reply:
This function only expects nn.Containers and otherwise smart pointers to be in here. Cereal will complain if you don't dereference the smart pointer as you haven't "registered the class" or something.

Reply:
Ah, the issue that without dereferencing, it expects an empty constructor, so it's a lot of boilerplate code for the user.